### ISSUE TITLE
Question for V4

### ISSUE THREAD

**Issue by EchterAlsFake** [Egsagon/PHUB#12]
So I see you are making a lot of progress on V4 and I want to ask if it's already ready to be used?
I want to start merging my GUI to v4, but I am not quite sure, because some exceptions are removed or not there and the Category object and the search filters are not there. 

I am just wondering :) 

Oh and, when I try to import phub I got this error:

Traceback (most recent call last):
  File "/home/asuna/PycharmProjects/Porn_Fetch/src/cli.py", line 3, in <module>
    from phub import Client, Quality, errors
  File "/home/asuna/PycharmProjects/Porn_Fetch/venv/lib/python3.11/site-packages/phub/__init__.py", line 8, in <module>
    from .core import Client
  File "/home/asuna/PycharmProjects/Porn_Fetch/venv/lib/python3.11/site-packages/phub/core.py", line 4, in <module>
    from . import utils
  File "/home/asuna/PycharmProjects/Porn_Fetch/venv/lib/python3.11/site-packages/phub/utils.py", line 9, in <module>
    from . import consts, locals 
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asuna/PycharmProjects/Porn_Fetch/venv/lib/python3.11/site-packages/phub/locals.py", line 7, in <module>
    from .objects import Param, NO_PARAM
ModuleNotFoundError: No module named 'phub.objects'

Not sure if it's because of my environment.

**Comment by Egsagon** [comment]
Hi

### Concerning PHUB4
I'd say the biggest features should be ok now, but i've not tested everything so there might be some bugs/typos/etc. I would suggest trying to implement it on a new branch for now.

I have not implemented: Images, Feed, Query counters, Tag comparisons, video categories (they are just parsed), user search, logs, download output display, and download backend presets (only default and ffmpeg are working).

### Concerning search filters
I've changed the way you input filters in Client.search.
All filters inherit from the Param object (from phub/objects/param.py) and you can concatenate them together, e.g.
```py
client.search('query', Category.FRENCH + Production.AMATEUR - Category.BBW)
```
In this example, we will search inside the french category for amateur productions, but only if they are not part of the BBW category. Note that due to PH limitations, we can only search in one Category, but we can exclude as many as we want (I have not asserted the edge cases yet).

Not sure if this is a good idea or not, i thought it would help while static typing.

Also, all of the constants objects have been moved to phub/locals.py so they are easier to import:
```py
import phub
from phub.locals import *

Category
Quality
Production
# etc.
```

### Concerning your import error
Thats a dumb error, because I have not even tried to install PHUB as a package yet.
I just have not included the object and modules folders to the package. I'll fix this in a bit.

**Comment by EchterAlsFake** [comment]
Okay thanks :)

**Comment by Egsagon** [comment]
Fixed the import error with 075886de2fc5c4a2b0b118ad858d1f6b00f252b5
